### PR TITLE
vault: piped input for ipa vault-add fails

### PR DESCRIPTION
**Pull request by flo-renaud**
An exception is raised when using echo "Secret123\n" | ipa vault-add myvault

This happens because the code is using (string).decode(sys.stdin.encoding)
and sys.stdin.encoding is None when the input is read from a pipe.
The fix is using the prompt_password method defined by Backend.textui,
which gracefully handles this issue.

https://bugzilla.redhat.com/show_bug.cgi?id=1445358

### CODE REVIEW

File: ipaclient/plugins/vault.py [hunk 1]
```diff
@@ -304,7 +279,8 @@ def forward(self, *args, **options):
                 password = password.rstrip('\n')
 
             else:
-                password = get_new_password()
+                password = self.api.Backend.textui.prompt_password(
```
Comment (Akasurde) [inline ipaclient/plugins/vault.py]: LGTM

Comment (stlaz) [inline ipaclient/plugins/vault.py]: Is this LGTM meant for the whole patch or just this one-liner? If the prior, please, don't add your LGTMs to single lines, if the latter, please, do complete reviews.

**Comment by stlaz** [comment]
@Akasurde: Don't add ACK label when the PR is not OK!
@flo-renaud: You will need to specify a ticket for this PR.

**Comment by flo-renaud** [comment]
@stlaz 
Thank you for the reminder. Commit msg updated with issue 6907

**Comment by stlaz** [comment]
Thank you for the brief action taken. Re-adding the ACK label.

File: ipaclient/plugins/vault.py (hunk 1 above)
Comment (Akasurde) [inline ipaclient/plugins/vault.py]: LGTM was for complete review. I tried the patch and worked for me, so add ACK and LGTM.

**Comment by MartinBasti** [comment]
master:

* d5c41ed4ad370c7d74296a830993a5bd3fd32e5f vault: piped input for ipa vault-add fails


ipa-4-5:

* c8ca0f89a68b5d57c56344fdeb12fd436976c726 vault: piped input for ipa vault-add fails

**Pull request closed without merging**